### DISCUSSION TITLE
Help - Font

### DISCUSSION THREAD

**Discussion by marcosgerene** [AvaloniaUI/Avalonia#16300] [Q&A]
Hi folks,

I need some help here. I am trying to use NotoSans as my font. So I added this to my assets.
![image](https://github.com/user-attachments/assets/b059636c-ab9d-4697-9a9c-7686736e4746)

It upgraded my project file:
```
<ItemGroup>
	...
	<AvaloniaResource Include="Assets\fonts\NotoSans-Italic.ttf" />
	<AvaloniaResource Include="Assets\fonts\NotoSans.ttf" />		
</ItemGroup>
```

Then I put it on my Applicatio.Resources and tried to use it as the font of my TextBlocks.

![image](https://github.com/user-attachments/assets/5dd49d18-6b59-48f8-8081-f223996dfaa5)

But I hit on this:

![image](https://github.com/user-attachments/assets/2b06646d-8964-4177-a48c-0a273f651931)

I have read some issues/discussions and documentation and tried Chat-GPT but it is not working.

----

I just renamed the folder from ```fonts``` to ```Fonts```and it worked, but...

![image](https://github.com/user-attachments/assets/7c4d2ab3-bc97-4287-8a8e-a1fa3b246ee2)

It is ugly, looks like thin font.

I tried:
```
<Style Selector="TextBlock">
	<Setter Property="FontFamily" Value="{DynamicResource NotoSansFont}"/>	
	<Setter Property="FontWeight" Value="Medium" />
</Style>

<Application.Resources>
   <FontFamily x:Key="NotoSansFont">avares://Panthera.Atualizador/Assets/Fonts#Noto Sans</FontFamily>
</Application.Resources>
```
Did not change anything.

I also tried:
```
<Application.Resources>
    <FontFamily x:Key="NotoSansFont">avares://Panthera.Atualizador/Assets/Fonts#Noto Sans Medium</FontFamily>
</Application.Resources>
```

and faced it again.
![image](https://github.com/user-attachments/assets/2b06646d-8964-4177-a48c-0a273f651931)

---

I just "discovered" the "medium" size and used the default font from Avalonia with medium and had the result I was looking for.

**Comment by stevemonaco** [comment]
You can check this discussion comment to see how Avalonia deals with custom font loading, particularly regarding the name (ie. the `#Noto Sans` part). https://github.com/AvaloniaUI/Avalonia/discussions/14298#discussioncomment-9552525

You may need to provide additional font properties to render it as expected (ie. `FontSize`, `FontWeight`, etc).
